### ISSUE TITLE
undefined method `table_name' for nil:NilClass

### ISSUE THREAD

**Issue by such** [rmosolgo/graphql-ruby#3581]
**Describe the bug**

When the graphqlAPI is called with an invalid cursor we can get this error:

`undefined method 'table_name' for nil:NilClass`

**Versions**

`graphql` version:
gem 'graphql', '~> 1.12.7'
gem 'graphql-pro', '1.17.8'

`rails` (or other framework):
gem 'rails', '~> 6.1.3.1'

**GraphQL schema**

Include relevant types and fields (in Ruby is best, in GraphQL IDL is ok).
Are you using [interpreter](https://graphql-ruby.org/queries/interpreter.html)? Any custom instrumentation, etc?

```ruby
class ApplicationSchema < GraphQL::Schema
  use GraphQL::Batch
  use GraphQL::Tracing::AppsignalTracing
  use GraphQL::Guard.new(
    not_authorized: lambda { |type, field|
      GraphQL::ExecutionError.new("Not authorized to access #{type}.#{field}")
    }
  )
  use GraphQL::Subscriptions::AnyCableSubscriptions, broadcast: true, default_broadcastable: true

  connections.add(ActiveRecord::Relation, GraphQL::Pro::PostgresStableRelationConnection)end
```

**GraphQL query**

Example GraphQL query and response (if query execution is involved)

```graphql
query {
  club(slug: \"bayern-munchen-munchen\") {
    activePlayers(after: \"WyJTdGFuacWhacSHIiwiZTc1MTMwMmYtNzRiMi00YTMwLTkxNDEtYWRhYTQ1NjYwMjkxIl0\") {
      nodes { ... }
    }
  }
}
```

**Steps to reproduce**

Pass an invalid cursor

**Expected behavior**

A proper error should be returned to the caller, or a proper exception should be raised so that it can be rescued cleanly

**Actual behavior**

A `NoMethodError: undefined method 'table_name' for nil:NilClass` is raised

<details>
<summary>Click to view exception backtrace</summary>

```
graphql-pro (1.17.8) lib/graphql/pro/relation_connection/condition.rb:46 block in build_condition |  
-- | --
graphql-pro (1.17.8) lib/graphql/pro/relation_connection/condition.rb:44 each |  
graphql-pro (1.17.8) lib/graphql/pro/relation_connection/condition.rb:44 each_with_index |  
graphql-pro (1.17.8) lib/graphql/pro/relation_connection/condition.rb:44 build_condition |  
graphql-pro (1.17.8) lib/graphql/pro/relation_connection/filter.rb:22 filter_relation |  
graphql-pro (1.17.8) lib/graphql/pro/stable_relation_connection.rb:100 load_nodes_by_value |  
graphql-pro (1.17.8) lib/graphql/pro/stable_relation_connection.rb:62 load_nodes |  
lib/graphql/pagination/relation_connection.rb:9 nodes |  
lib/graphql/pagination/connection.rb:147 edge_nodes |  
lib/graphql/types/relay/connection_behaviors.rb:138 nodes |  
lib/graphql/schema/field.rb:716 public_send |  
lib/graphql/schema/field.rb:716 block in public_send_field |  
lib/graphql/schema/field.rb:757 block in with_extensions |  
lib/graphql/schema/field.rb:787 block (2 levels) in run_extensions_before_resolve |  
lib/graphql/schema/field.rb:790 run_extensions_before_resolve |  
lib/graphql/schema/field.rb:787 block in run_extensions_before_resolve |  
lib/graphql/schema/field_extension.rb:50 resolve |  
lib/graphql/schema/field.rb:780 run_extensions_before_resolve |  
lib/graphql/schema/field.rb:756 with_extensions |  
lib/graphql/schema/field.rb:696 public_send_field |  
lib/graphql/schema/field.rb:609 block in resolve |  
lib/graphql/schema.rb:115 after_lazy |  
lib/graphql/schema/field.rb:607 resolve |  
lib/graphql/execution/interpreter/runtime.rb:272 block (4 levels) in evaluate_selection_with_args |  
lib/graphql/tracing.rb:81 call_tracers |  
lib/graphql/tracing.rb:83 block in call_tracers |  
graphql-guard (2.0.0) lib/graphql/guard.rb:40 block in trace |  
graphql-guard (2.0.0) lib/graphql/guard.rb:68 ensure_guarded |  
graphql-guard (2.0.0) lib/graphql/guard.rb:40 trace |  
lib/graphql/tracing.rb:83 call_tracers |  
lib/graphql/tracing.rb:83 block in call_tracers |  
lib/graphql/tracing/platform_tracing.rb:52 block in trace |  
lib/graphql/tracing/appsignal_tracing.rb:19 block in platform_trace |  
lib/appsignal/helpers/instrumentation.rb:525 instrument |  
lib/graphql/tracing/appsignal_tracing.rb:18 platform_trace |  
lib/graphql/tracing/platform_tracing.rb:51 trace |  
lib/graphql/tracing.rb:83 call_tracers |  
lib/graphql/tracing.rb:67 trace |  
lib/graphql/execution/interpreter/runtime.rb:271 block (3 levels) in evaluate_selection_with_args |  
lib/graphql/query.rb:362 block in with_error_handling |  
lib/graphql/execution/errors.rb:107 with_error_handling |  
lib/graphql/query.rb:361 with_error_handling |  
lib/graphql/execution/interpreter/runtime.rb:270 block (2 levels) in evaluate_selection_with_args |  
lib/graphql/execution/interpreter/runtime.rb:433 resolve_with_directives |  
lib/graphql/execution/interpreter/runtime.rb:267 block in evaluate_selection_with_args |  
lib/graphql/execution/interpreter/runtime.rb:517 after_lazy |  
lib/graphql/execution/interpreter/runtime.rb:214 evaluate_selection_with_args |  
lib/graphql/execution/interpreter/runtime.rb:202 evaluate_selection |  
lib/graphql/execution/interpreter/runtime.rb:142 block (2 levels) in evaluate_selections |  
lib/graphql/dataloader/null_dataloader.rb:16 append_job |  
lib/graphql/execution/interpreter/runtime.rb:141 block in evaluate_selections |  
lib/graphql/execution/interpreter/runtime.rb:140 each |  
lib/graphql/execution/interpreter/runtime.rb:140 evaluate_selections |  
lib/graphql/execution/interpreter/runtime.rb:385 block in continue_field |  
lib/graphql/execution/interpreter/runtime.rb:517 after_lazy |  
lib/graphql/execution/interpreter/runtime.rb:379 continue_field |  
lib/graphql/execution/interpreter/runtime.rb:426 continue_field |  
lib/graphql/execution/interpreter/runtime.rb:281 block (3 levels) in evaluate_selection_with_args |  
lib/graphql/execution/interpreter/runtime.rb:517 after_lazy |  
lib/graphql/execution/interpreter/runtime.rb:507 block in after_lazy |  
lib/graphql/execution/lazy.rb:41 value |  
lib/graphql/execution/interpreter/resolve.rb:44 resolve |  
lib/graphql/execution/interpreter/resolve.rb:10 block in resolve_all |  
lib/graphql/dataloader/null_dataloader.rb:16 append_job |  
lib/graphql/execution/interpreter/resolve.rb:10 resolve_all |  
lib/graphql/execution/interpreter.rb:98 block in sync_lazies |  
lib/graphql/tracing.rb:81 call_tracers |  
lib/graphql/tracing.rb:83 block in call_tracers |  
graphql-guard (2.0.0) lib/graphql/guard.rb:42 trace |  
lib/graphql/tracing.rb:83 call_tracers |  
lib/graphql/tracing.rb:83 block in call_tracers |  
lib/graphql/tracing/platform_tracing.rb:26 block in trace |  
lib/graphql/tracing/appsignal_tracing.rb:19 block in platform_trace |  
lib/appsignal/helpers/instrumentation.rb:525 instrument |  
lib/graphql/tracing/appsignal_tracing.rb:18 platform_trace |  
lib/graphql/tracing/platform_tracing.rb:25 trace |  
lib/graphql/tracing.rb:83 call_tracers |  
lib/graphql/tracing.rb:67 trace |  
lib/graphql/execution/interpreter.rb:97 sync_lazies |  
lib/graphql/execution/interpreter.rb:55 finish_multiplex |  
lib/graphql/execution/multiplex.rb:111 block in run_as_multiplex |  
lib/graphql/dataloader/null_dataloader.rb:16 append_job |  
lib/graphql/execution/multiplex.rb:110 run_as_multiplex |  
lib/graphql/execution/multiplex.rb:63 block (2 levels) in run_queries |  
lib/graphql/execution/multiplex.rb:208 block in instrument_and_analyze |  
lib/graphql/execution/instrumentation.rb:29 block (2 levels) in apply_instrumenters |  
lib/graphql/execution/instrumentation.rb:46 block (2 levels) in each_query_call_hooks |  
lib/graphql/execution/instrumentation.rb:41 each_query_call_hooks |  
lib/graphql/execution/instrumentation.rb:45 block in each_query_call_hooks |  
lib/graphql/execution/instrumentation.rb:72 call_hooks |  
lib/graphql/execution/instrumentation.rb:44 each_query_call_hooks |  
lib/graphql/execution/instrumentation.rb:27 block in apply_instrumenters |  
lib/graphql/execution/instrumentation.rb:72 call_hooks |  
lib/graphql/execution/instrumentation.rb:26 apply_instrumenters |  
lib/graphql/execution/multiplex.rb:187 instrument_and_analyze |  
lib/graphql/execution/multiplex.rb:62 block in run_queries |  
lib/graphql/tracing.rb:81 call_tracers |  
lib/graphql/tracing.rb:83 block in call_tracers |  
graphql-guard (2.0.0) lib/graphql/guard.rb:42 trace |  
lib/graphql/tracing.rb:83 call_tracers |  
lib/graphql/tracing.rb:83 block in call_tracers |  
lib/graphql/tracing/platform_tracing.rb:26 block in trace |  
lib/graphql/tracing/appsignal_tracing.rb:19 block in platform_trace |  
lib/appsignal/helpers/instrumentation.rb:525 instrument |  
lib/graphql/tracing/appsignal_tracing.rb:18 platform_trace |  
lib/graphql/tracing/platform_tracing.rb:25 trace |  
lib/graphql/tracing.rb:83 call_tracers |  
lib/graphql/tracing.rb:67 trace |  
lib/graphql/execution/multiplex.rb:60 run_queries |  
lib/graphql/execution/multiplex.rb:50 run_all |  
lib/graphql/schema.rb:1682 multiplex |  
lib/graphql/schema.rb:1653 execute |  
app/controllers/graphql_controller.rb:41 execute_query
```

</details>

**Comment by rmosolgo** [comment]
Thanks for the detailed report! I agree it should fail properly, and I think I have everything I need here to sort it out and release a fix. I'll follow up here when it's ready!

**Comment by rmosolgo** [comment]
I've just released graphql-pro 1.18.2 which _should_ handle this case nicely. Thanks again for the detailed report, and please let me know if you run into any more trouble after updating!
